### PR TITLE
Fix healthcheck crash on invalid kubeconfig

### PR DESCRIPTION
**Pull request by baijum**
Ref. https://issues.redhat.com/browse/APPSVC-1372

### CODE REVIEW

**Review by Kartikey-star — APPROVED** [review]
lgtm, Checking whether the config being returned is empty can  be an additional check  after https://github.com/primaza/primaza/pull/210/ is merged or either this PR is merged. This condition and status updates can be added in `extractClusterContextRESTConfig` method. WDYT?

File: controllers/clusterenvironment_controller.go [hunk 1]
```diff
@@ -278,6 +278,20 @@ func (r *ClusterEnvironmentReconciler) MonitorHealth(ctx context.Context, ns str
 					}
 				}
 			}
+			if cfg == nil {
+				l.Error(errors.New("Empty conig"), "kubeclient config is empty")
```
Comment (filariow) [inline controllers/clusterenvironment_controller.go:282]:
```suggestion
				l.Info("Not running healthchecks for ClusterEnvironment: empty kubeconfig from ClusterContext Secret", "ClusterEnvironment", ce.Name)
```

File: controllers/clusterenvironment_controller.go [hunk 2]
```diff
@@ -278,6 +278,20 @@ func (r *ClusterEnvironmentReconciler) MonitorHealth(ctx context.Context, ns str
...
+				c := workercluster.ConnectionStatus{
+					State:   primazaiov1alpha1.ClusterEnvironmentStateOffline,
+					Reason:  ErrorDuringHealthCheckReason,
+					Message: "kubeclient config is empty",
+				}
+				r.updateClusterEnvironmentStatus(ctx, ce, c)
+				if err := r.Client.Status().Update(ctx, ce); err != nil {
+					l.Error(err, "error updating cluster environment status", "status", ce.Status)
+				}
+				time.Sleep(time.Duration(hci) * time.Second)
```
Comment (filariow) [inline controllers/clusterenvironment_controller.go:292]: Is this needed? I'd say we can continue directly with next ClusterEnvironment in the list and sleep at L333

```suggestion
```

File: controllers/clusterenvironment_controller.go [hunk 3]
```diff
@@ -278,6 +278,20 @@ func (r *ClusterEnvironmentReconciler) MonitorHealth(ctx context.Context, ns str
 					}
 				}
 			}
+			if cfg == nil {
```
Comment (filariow) [inline controllers/clusterenvironment_controller.go:281]: we may handle this check in `clustercontext.GetClusterRESTConfig` and return an error from there. WDYT?

**Review by filariow — CHANGES_REQUESTED** [review]
some ideas

File: controllers/clusterenvironment_controller.go (hunk 3 above)
Comment (baijum) [reply controllers/clusterenvironment_controller.go:281]: There are other places where this function is called where empty config is compatible. So, I would prefer to retain the current code. I will work on the other suggestions.

File: controllers/clusterenvironment_controller.go (hunk 2 above)
Comment (baijum) [reply controllers/clusterenvironment_controller.go]: Done.

File: controllers/clusterenvironment_controller.go (hunk 1 above)
Comment (baijum) [reply controllers/clusterenvironment_controller.go]: Done.

File: controllers/clusterenvironment_controller.go (hunk 3 above)
Comment (filariow) [reply controllers/clusterenvironment_controller.go:281]: Right, good point!

**Comment by baijum** [comment]
> lgtm, Checking whether the config being returned is empty can be an additional check after #210 is merged or either this PR is merged. This condition and status updates can be added in `extractClusterContextRESTConfig` method. WDYT?

@Kartikey-star 

Now PR #210 has been merged. I have updated this PR to call  `extractClusterContextRESTConfig` method. Since empty config is compatible in other places where that function is called, I have kept the check outside the function.

File: controllers/clusterenvironment_controller.go [hunk 4]
```diff
@@ -735,7 +748,7 @@ func (r *ClusterEnvironmentReconciler) CreateServiceCatalog(ctx context.Context,
 }
 
 func (r *ClusterEnvironmentReconciler) finalizeClusterEnvironmentInNamespaces(ctx context.Context, ce *primazaiov1alpha1.ClusterEnvironment) error {
-	kcfg, err := clustercontext.GetClusterRESTConfig(ctx, r.Client, ce.Namespace, ce.Spec.ClusterContextSecret)
+	kcfg, err := r.retrieveClusterContextSecret(ctx, ce)
```
Comment (baijum) [inline controllers/clusterenvironment_controller.go:751]: This is an unrelated change. I made this change based on PR #210
Let me know if I should send a separate PR.

**Review by filariow — APPROVED** [review]
lgtm
